### PR TITLE
Issue #458 : Fixing serializer for JSR107

### PR DESCRIPTION
**Pull request by rishabhmonga**

### CODE REVIEW

File: 107/src/test/java/org/ehcache/jsr107/LongSerializer.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    LOG.info("serializing {}", object);
+    ByteBuffer byteBuffer = ByteBuffer.allocate(8);
+    byteBuffer.putLong(object);
+    byteBuffer.flip();
+    return byteBuffer;
+  }
+
+  @Override
+  public Long read(ByteBuffer binary) throws IOException, ClassNotFoundException {
+//    binary.flip();
```
Comment (chrisdennis) [inline 107/src/test/java/org/ehcache/jsr107/LongSerializer.java]: delete this line please

File: 107/src/test/java/org/ehcache/jsr107/LongSerializer.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    long l = binary.getLong();
+    LOG.info("deserialized {}", l);
```
Comment (chrisdennis) [inline 107/src/test/java/org/ehcache/jsr107/LongSerializer.java]: No need to log here: `return binary.getLong();`

File: 107/src/test/java/org/ehcache/jsr107/LongSerializer.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+public class LongSerializer implements Serializer<Long> {
+  private static final Logger LOG = LoggerFactory.getLogger(LongSerializer.class);
+  private static final Charset CHARSET = Charset.forName("US-ASCII");
+
+  public LongSerializer(ClassLoader classLoader) {
+  }
+
+  @Override
+  public ByteBuffer serialize(Long object) throws IOException {
+    LOG.info("serializing {}", object);
```
Comment (chrisdennis) [inline 107/src/test/java/org/ehcache/jsr107/LongSerializer.java]: No need to log here.

File: 107/src/test/java/org/ehcache/jsr107/StringSerializer.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    byteBuffer.flip();
+    return byteBuffer;
+  }
+
+  @Override
+  public String read(ByteBuffer binary) throws IOException, ClassNotFoundException {
+    byte[] bytes = new byte[binary.remaining()];
+    binary.get(bytes);
+    String s = new String(bytes, CHARSET);
+    LOG.info("deserialized {}", s);
```
Comment (chrisdennis) [inline 107/src/test/java/org/ehcache/jsr107/StringSerializer.java]: no need for logging here

File: 107/src/test/java/org/ehcache/jsr107/StringSerializer.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+public class StringSerializer implements Serializer<String> {
+  private static final Logger LOG = LoggerFactory.getLogger(StringSerializer.class);
+  private static final Charset CHARSET = Charset.forName("US-ASCII");
+
+  public StringSerializer(ClassLoader classLoader) {
+  }
+
+  @Override
+  public ByteBuffer serialize(String object) throws IOException {
+    LOG.info("serializing {}", object);
```
Comment (chrisdennis) [inline 107/src/test/java/org/ehcache/jsr107/StringSerializer.java]: no need for logging here

**Comment by rishabhmonga** [comment]
test this please

**Comment by AbfrmBlr** [comment]
this needs a rebase

**Comment by lorban** [comment]
Apparently, this needs to be rebased again

**Pull request merged**